File: src/checkers-api/Models/GameModels/Piece.cs
namespace checkers_api.Models.GameModels;

public class Piece
{
    public string OwnerId { get; }
    public PieceState State { get; private set; }

    public Piece(string ownerId)
    {
        ArgumentNullException.ThrowIfNull(ownerId);

        OwnerId = ownerId;
        State = PieceState.Regular;
    }

    public void KingPiece()
    {
        State = PieceState.King;
    }
}

File: src/checkers-client/src/game-models/piece.ts
export default class Piece {
  public readonly color: string;
  public readonly ownerId: string;
  public readonly state: string;

  constructor(color: number, ownerId: string, state: number) {
    this.color = color == 0 ? "white" : "black";
    this.ownerId = ownerId;
    this.state = state == 0 ? "regular" : "King";
  }
}
